src/mono/mono/mini/aot-compiler.c

```diff
@@ -10722,46 +10766,349 @@ mono_aot_patch_info_dup (MonoJumpInfo* ji)
 	return res;
 }
 
+static int driver_log_file_fd = -1;
+
+typedef enum {
+	CROSS_COMPILER_COMMAND,
```

Comment:
Wouldn't it be possible to just replace this with a generic 'exec_command' facility ?

Reply:
Wanted to have a more structural way to describe supported commands making it simpler for clients to quickly identified the commands that are requested to be executed. Also, since several tools have different names on different platforms, having a way to type it also makes it simpler for the clients to identify what things its supposed to executed. The pace of adding new commands to AOT compiler is also very slow, so don't anticipate to much change here either.